### PR TITLE
Bug 1346251 Prefer newer builtin methods over lodash

### PR DESCRIPTION
**Pull request by stanleyye**
Couldn't find too many builtin methods > lodash. There were a some lodash methods such as defaults and deepDefaults but its a bit different than something like Object.create().

### CODE REVIEW

File: test/integration/encrypted_env_variables_test.js [hunk 1]
```diff
@@ -37,7 +37,7 @@ suite('encrypted private env variables', () => {
 
   function getEncryptedEnvPayload(payloadData) {
 
-    return Promise.all(_.map(payloadData, (data) => {
+    return Promise.all(payloadData.map((data) => {
```
Comment (gregarndt) [inline test/integration/encrypted_env_variables_test.js]: Could you have a look at the test files where you removed lodash and see if the import of lodash could be removed?

Comment (stanleyye) [inline test/integration/encrypted_env_variables_test.js]: sure, it seems like I missed that

File: src/lib/private_key.js [hunk 2]
```diff
@@ -95,7 +94,7 @@ PrivateKey.prototype = {
 
     // For each encrypted variable, create a promise and wait for all
     // promises to complete
-    return Promise.all(_.map(taskPayload.encryptedEnv, function(encryptedVar) {
+    return Promise.all(taskPayload.encryptedEnv.map(function(encryptedVar) {
```
Comment (gregarndt) [inline src/lib/private_key.js]: This will thrown an error if taskPayload.encryptedEnv is undefined

File: src/lib/features/artifacts.js [hunk 3]
```diff
@@ -181,14 +180,14 @@ export default class Artifacts {
     if (typeof artifacts !== 'object') return;
 
     // Upload all the artifacts in parallel.
-    await Promise.all(_.map(artifacts, (value, key) => {
+    await Promise.all(artifacts.map((value, key) => {
```
Comment (gregarndt) [inline src/lib/features/artifacts.js]: artifacts is an object, which does not have a map() method.

File: src/lib/features/artifacts.js [hunk 4]
```diff
@@ -181,14 +180,14 @@ export default class Artifacts {
...
       return this.uploadArtifact(taskHandler, key, value).catch((err) => {
         errors[key] = err;
       });
     }));
 
     if (Object.keys(errors).length) {
-      _.map(errors, (value, key) => {
+      errors.map((value, key) => {
```
Comment (gregarndt) [inline src/lib/features/artifacts.js]: errors is an object, which does not have a map() method

**Comment by stanleyye** [comment]
Let me get back to you on this later, I was having trouble running the tests.

**Comment by gregarndt** [comment]
Running tests can be problematic locally if you do not have TaskCluster credentials.  It might be easiest to make changes and push them up to the PR to let CI capture them.

**Comment by stanleyye** [comment]
sweet, that would be ideal. Is there any way for CI to automatically capture the changes? It seems like you have to close and re-open the PR? As well how do you know what errors there are from the graphs? I can't find the page where the test errors are displayed.

**Comment by gregarndt** [comment]
It's not all the time where a close/open triggers CI, but sometimes it helps to kick start it if it didn't automatically start for some reason.  Let's see what happens the next time you push.  If things work well, you should see a comment like: https://github.com/taskcluster/docker-worker/pull/287#issuecomment-294929948

that link will take you to a TaskCluster group inspector, which can a little confusing to get to the results you need (improvements to it are in the pipeline).  From there you will have a list of tasks, you can select any one of them, then click the menu for "Runs", select the latest run, and click the Logs link.

File: src/lib/features/artifacts.js (hunk 3 above)
Comment (stanleyye) [inline src/lib/features/artifacts.js]: is it best to just leave this part alone? I was thinking of Object.keys(artifacts)..map(function(key, index) { .... }) but then you have to call the accessor like artifacts[key] for the value. With lodash, its much more simpler. What do you think?

**Comment by stanleyye** [comment]
hey @gregarndt, it seems the tests fail with no changes made?

**Comment by gregarndt** [comment]
hey @stanleyye , sorry that you're seeing failures.  Unfortunately some of these are flakey and result in some false errors.  If you submit your recent changes I'm happy to sort through the errors and run them locally to see what is truly a failure.  This isn't ideal, I admit, but it'll work until time is invested in making the tests less error prone.  Thanks for your persistence and patience!

**Comment by stanleyye** [comment]
hi @gregarndt, could you try testing my latest changes?

**Comment by stanleyye** [comment]
it looks like everything passed? not sure if some other small changes i made were worthwhile.

**Comment by gregarndt** [comment]
Sorry for the long delay on this, but you're right, it looks like things are ok! Thanks for your patience on this PR and for your help getting it landed!

**Pull request merged**